src/main.rs

```diff
@@ -464,10 +466,46 @@ enum TestOutcome {
 }
 
 impl Toolchain {
+    fn default_nightly() -> Option<Date<Utc>> {
```

Comment:
We could cache calls to this function :)

Reply:
But we don't :joy:, it's not really important given the order of magnitudes of things executing here. We could use `lazy_static` if we were going to do so.